reflex: add page (#4884)
tldr-pages/tldr: pages/common/reflex.md

# reflex

> > Reflex is a small tool to watch a directory and rerun a command when certain files change.

Reviewer: Perhaps this will read better?
```suggestion
> Tool to watch a directory and rerun a command when certain files change.
```

> More information: <https://github.com/cespare/reflex>.

- Rebuild with `make` if any file changes:

`reflex make`

> - Compile and run Go application if any .go file changes:

Reviewer: Perhaps this will read better?
```suggestion
- Compile and run a Go application if any `.go` files change:
```

Author: @sbrl great suggestion!

Reviewer 2: @pjvds you should add the back ticks to highlight `.go` extension:
 ```suggestion
- Compile and run Go application if any `.go` file changes:
```

`reflex -r '{{\.go$}}' {{go run .}}`

- Ignore a directory when watching for changes:

> `reflex -R '{{^dir/}}' {{command}}`

Reviewer 2: Can you apply the long argument format for `-R` and `-r` examples?

Reviewer 2: ```suggestion
`reflex --inverse-regex='{{^dir/}}' {{command}}`
```

- Run command when reflex starts and restarts on file changes:

> `reflex -s go test`

Reviewer 2: Please use long argument format.

Author: @navarroaxel what do you mean by long argument format?

Reviewer 2: ```suggestion
`reflex --start-service=false go test`
```

Author: @navarroaxel thanks for the clearification. I looked at the style guide, tldr page and some others (zip, ln, watch, entr) and they all favor shorthand arguments over the long arguments.

I don't mind changing my pull request according to your feedback, I just want to make sure it's the right direction.

Reviewer 2: We prefer the use of long arguments if possible, but we have exceptions for them when it's really uncommon the use of long arguments, e.g.  `pacman -Syu`, etc

Author: Thanks for making that clear. I updated my changes accordingly.

> - Substitude what file changes:

Reviewer: Perhaps this will read better?
```suggestion
- Substitute the filename that changed in:
```
